### PR TITLE
[Datasets] Split test_dataset.py completely into test_dataset_{consumption,map,all_to_all}.py

### PR DESCRIPTION
**Pull request by c21**
## Why are these changes needed?

End-to-end runtime of `test_dataset.py` is around test timeout threshold (15 minutes for bazel large test). This PR is to split `test_dataset.py` completely into 4 separate test files:

* `test_dataset_consumption.py`: the tests for consumption APIs and all misc tests
* `test_dataset_map.py`: the tests for map-like transformation APIs
* `test_dataset_all_to_all.py`: the tests for all-to-all transformation APIs
* `test_dataset_ecosystem.py`: the tests for from/to_modin/dask APIs

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/data/tests/test_dataset_integration.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
+import numpy as np
```
Comment (ericl) [inline python/ray/data/tests/test_dataset_integration.py:1]: Instead of integration, could we call this test_dataset_ecosystem?

**Review by ericl — APPROVED** [review]

File: python/ray/data/tests/test_dataset_map.py [hunk 2]
```diff
@@ -0,0 +1,920 @@
+import itertools
```
Comment (jianoaix) [inline python/ray/data/tests/test_dataset_map.py:1]: Maybe a file level string to explain what this is covering (as what PR description says)?

**Review by jianoaix — APPROVED** [review]

File: python/ray/data/tests/test_dataset_map.py (hunk 2 above)
Comment (c21) [reply python/ray/data/tests/test_dataset_map.py:1]: Yes make sense. I am thinking of adding a readme in test directory, to explain each file. I want to unbreak CI as soon as possible, and avoid merge conflict with other PR. Can we do it as followup?

**Review by clarkzinzow — APPROVED** [review]
LGTM! `test_dataset_consumption.py` is a bit of a misnomer since it contains a lot of non-consumption tests (e.g. `from_items`, `range_table`, global aggregations, etc.), but we work on further splitting that out as a follow-up.

**Pull request merged**